src/redux/stores/balances/balancesHelpers.ts

```diff
@@ -54,21 +38,53 @@ export const getOwnedTokenIdsOfWallet = async (
     const transferFilter = collectionContract.filters.Transfer(null, walletAddress);
 
     const events: Event[] = await collectionContract.queryFilter(transferFilter, 0);
+
     /* get token ids from past events */
     const foundTokenIds: BigNumber[] = events.map(e => e.args?.at(2));
 
-    return getUniqueTokensForWallet(foundTokenIds, collectionContract, walletAddress);
+    /* get unique values */
+    const uniqueTokenIds = [...new Set(foundTokenIds)];
+
+    /* get owners of tokens */
+    const tokenOwners: string[] = await Promise.all(
+      uniqueTokenIds.map(tokenId => collectionContract.ownerOf(tokenId)),
+    );
+
+    /* get only the owned token ids */
+    const ownedTokenIds = uniqueTokenIds.filter((_, index) => tokenOwners[index] === walletAddress);
+
+    /* return sorted array of numbers */
+    return ownedTokenIds
+      .map(t => t.toNumber())
+      .sort((a, b) => a - b);
   }
 
   if (isErc1155) {
     const collectionContract = new ethers.Contract(collectionToken, erc1155Contract.abi, provider);
     const transferFilter = collectionContract.filters.TransferSingle(null, null, walletAddress);
 
     const events: Event[] = await collectionContract.queryFilter(transferFilter, 0);
+
     /* get token ids from past events */
     const foundTokenIds: BigNumber[] = events.map(e => e.args?.at(3));
 
-    return getUniqueTokensForWallet(foundTokenIds, collectionContract, walletAddress);
+    /* get unique values */
+    const uniqueTokenIds = [...new Set(foundTokenIds)];
```

Comment:
Because this trick doesn't work on objects. You should convert it to number first:

`const uniqueTokenIds = [...new Set(foundTokenIds.map(id => id.toNumber()))];`